pdf/model/optimize/image_ppi.go

```diff
@@ -0,0 +1,202 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package optimize
+
+import (
+	"fmt"
+	"image"
+	"math"
+
+	"github.com/unidoc/unidoc/pdf/contentstream"
+	. "github.com/unidoc/unidoc/pdf/core"
+	"github.com/unidoc/unidoc/pdf/model"
+	"golang.org/x/image/draw"
+)
+
+// ImagePPI optimizes images.
```

Comment:
Maybe make the comment a bit more descriptive. E.g.
ImagePPI optimizes images by scaling images such that the PPI (pixels per inch) is never higher than `ImageUpperPPI`.

And // TODO: Add support for inline images.